main/main.c

```diff
@@ -102,6 +140,72 @@ rte_strdup(const char *type, const char *s)
 	return strcpy(res, s);
 }
 
+FILE *log_file = NULL;
+struct rte_timer log_timer;
+
+static int
+cleanup_log(void)
+{
+	if (log_file != NULL) {
+		fclose(log_file);
+		log_file = NULL;
+	}
+
+	rte_free(log_base_dir);
+	rte_free(log_file_name_format);
+
+	return 0;
+}
+
+int
+gatekeeper_log_init(void)
+{
+	int ret;
+	time_t now;
+	struct tm *time_info;
+	char log_file_name[100];
+	char log_file_path[100];
+	FILE *new_log_file = NULL;
```

Comment:
The first use of `new_log_file` is an assignment, so we don't need to initialize it here.